src/main/java/com/auctionapp/api/model/entities/EntityWithUUID.java

```diff
@@ -0,0 +1,25 @@
+package com.auctionapp.api.model.entities;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+
+import org.hibernate.annotations.Type;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@MappedSuperclass
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public abstract class EntityWithUUID implements Serializable {
```

Comment:
Why do we need this class? Can't all entities just define their own id? I might be missing something, so please do tell me if you believe this should stay.